src/service_daemon.rs

```diff
@@ -1199,20 +1232,24 @@ impl Zeroconf {
         self.my_services.insert(service_fullname, info);
     }
 
-    /// Sends out annoucement of `info` on every valid interface.
-    /// Returns the list of interface IPs that sent out the annoucement.
+    /// Sends out announcement of `info` on every valid interface.
+    /// Returns the list of interface IPs that sent out the announcement.
     fn send_unsolicited_response(&self, info: &ServiceInfo) -> Vec<IpAddr> {
         let mut outgoing_addrs = Vec::new();
-        let mut subnet_set: HashSet<u128> = HashSet::new();
+        // Send the announcement on one interface per ip version.
+        let mut multicast_notification_trackers = HashSet::new();
 
         for (_, intf_sock) in self.intf_socks.iter() {
-            let subnet = ifaddr_subnet(&intf_sock.intf.addr);
-            if !intf_sock.intf.is_link_local() && subnet_set.contains(&subnet) {
-                continue; // no need to send again in the same subnet.
+            if intf_sock.as_multicast_notification_tracker().is_some()
```

Comment:
Wouldn't it be better to extract the value from Option ? like:
```rust
if let Some(tracker) = intf_sock.as_multicast_notification_tracker() {
  if multicast_notification_trackers.contains(tracker) {
     continue;
  }
}
```

Reply:
Ah yes, looks way more like using `rust` :)